presto-main/src/main/java/io/prestosql/execution/CreateMaterializedViewTask.java

```diff
@@ -86,7 +89,7 @@ public String explain(CreateMaterializedView statement, List<Expression> paramet
 
         String sql = getFormattedSql(statement.getQuery(), sqlParser);
 
-        Analysis analysis = new Analyzer(session, metadata, sqlParser, groupProvider, accessControl, Optional.empty(), parameters, parameterLookup, stateMachine.getWarningCollector())
+        Analysis analysis = new Analyzer(session, metadata, sqlParser, groupProvider, accessControl, Optional.empty(), parameters, parameterLookup, stateMachine.getWarningCollector(), statsCalculator)
```

Comment:
It's weird that Analyzer needs a stats calculator. In any case, why not provide a NOOP here?